snapcraft/plugins/v2/autotools.py

```diff
@@ -72,6 +72,8 @@ def _get_configure_command(self) -> str:
 
     def get_build_commands(self) -> List[str]:
         return [
+            "[ ! -f ./configure ] && [ -f ./autogen.sh ] && env NOCONFIGURE=1 ./autogen.sh",
```

Comment:
From my limited reading, it sounds like a project would generally use `autogen.sh` script, `bootstrap` script, or invoke `autoreconf`.  autoreconf after autogen might override the results of the autogen.sh / boostrap?

FWIW according to https://www.gnu.org/software/automake/faq/autotools-faq.html#What-does-_002e_002fbootstrap-or-_002e_002fautogen_002esh-do_003f  `bootstrap` is preferred over 'autogen.sh' (for the case both are present?)

Reply:
we discussed this online. bootstrap or autogen.sh will generate should configure script at the end of their run so these three lines are mutually exclusive.